docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,288 @@ keywords:
+In this guide you will learn how to provide an extension point so that other app plugins can add their extensions to your plugin.
 
-:::note
+## What is an extension point?
 
-Extension points living in core Grafana must start with `grafana/` and extension points living in plugins must have IDs starting with `plugins/` followed by the plugin ID, for example, `plugins/<PLUGIN_ID>/`.
+An extension point is a place in the UI which you make extendable by other plugins using extensions. These extensions can be either be links or simple React components (that can basically impelement anything). It's up to you how you are displaying the handling or displaying the extensions, we have some examples below.
 
-:::
+You can also share contextual information with the extensions using a `context` object or component props (see the following examples).
 
-The second thing you need to consider is how to design the UI of the extension point so it supports a scenario where multiple extensions are being added without breaking the UI.
+### Requirements
 
-Finally, consider if there is any information from the current view that should be shared with the extensions added to the extension point. It could be information from the current view that could let the extending plugin prefill values or other data in the functionality being added via the extension.
+- **an extension point ID** _(string)_
+  - In case it's an extension point in core Grafana, it must start with `grafana/`
+  - In case it's inside a plugin, it must start with `plugin/<PLUGIN_ID>/`
+  - It must be unique
+- **an app plugin** - _currently only app plugins can create extension points_
 
-## Create an extension point
+:::note
+
+When designing the UI make sure the extension point supports a scenario where multiple extensions can be added without breaking the UI. Also consider if there is any information from the current view that should be shared with the extensions added to the extension point. It could be information from the current view that could let the extending plugin pre-fill values or other data in the extension's functionality.
 
-Use the `getPluginLinkExtensions` method in `@grafana/runtime` to create an extension point within your plugin.
+:::
+
+## Create an extension point
 
 :::danger
 
 When you create an extension point in a plugin, you create a public interface for other plugins to interact with. Changes to the extension point ID or its context could break any plugin that attempts to register a link inside your plugin.
 
 :::
 
-The `getPluginLinkExtensions` method takes an object consisting of the `extensionPointId`, which must begin `plugins/<PLUGIN_ID>`, and any contextual information that you want to provide. The `getPluginLinkExtensions` method returns a list of extension links that your program can then loop over.
+You can easily create an extension point using the following functions (they live in `@grafana/runtime`) to fetch extensions for a certain extension point ID:
```

Comment:
```suggestion
You can easily create an extension point using the following functions from `@grafana/runtime` to fetch extensions for a certain extension point ID:
```
Avoids the use of brackets

Reply:
Its also somewhat brief as an introduction. "Create an extension point" sounds like I'm going to get a tutorial, but instead we've got more of an API doc with examples. Additionally we say functions plural, but then have 2 "sections" with one of them a deprecated method.

I don't object to the format, but I think we may need to expand this sentence. Happy to follow up with more suggestions if you and agree and would like some help with this

Reply:
Yeah, we could probably add more content here, let's try to make it better together 👍 

> but instead we've got more of an API doc with examples

True, although maybe if we start reorganising the structure of the docs soon ("Reference" and How to ..." are separate sections), then it will be good for the API reference.

> Additionally we say functions plural, but then have 2 "sections" with one of them a deprecated method.

Hmm, I guess as long as we don't remove the getter functions plural is legit?